tests/contracts/TestRulesFactory.sol

```diff
@@ -54,9 +95,66 @@ contract TestRulesFactory is Stakable {
         else if (rule.eq("SELFDESTRUCT")) {
             coin.destruct();
         }
+        else if (rule.eq("CALL_undeployed_contract")) {
+            address(100100).call("");
+        }
+        else if (rule.eq("CALLCODE_undeployed_contract")) {
+            assembly {
+                let x := mload(0x40)
+                mstore(x, 0xffffffff)
+                mstore(add(x,0x04), 0)
+                mstore(add(x,0x24), 0)
+                let res := callcode(5000, 100200, 0, x, 0x44, add(x,0x80), 0)
+            }
+        }
+        else if (rule.eq("DELEGATECALL_undeployed_contract")) {
+            address(100300).delegatecall("");
+        }
+        else if (rule.eq("STATICCALL_undeployed_contract")) {
+            address(100400).staticcall("");
+        }
+        else if (rule.eq("EXTCODESIZE_undeployed_contract")) {
+            emit Uint(address(100500).code.length);
+        }
+        else if (rule.eq("EXTCODEHASH_undeployed_contract")) {
+            emit Uint(uint256(address(100600).codehash));
+        }
+        else if (rule.eq("EXTCODECOPY_undeployed_contract")) {
+            bytes memory o_code;
+            assembly {
+                // retrieve the size of the code, this needs assembly
+                // let size := extcodesize(_addr)
+                let size := 0
+                // allocate output byte array - this could also be done without assembly
+                // by using o_code = new bytes(size)
+                o_code := mload(0x40)
+                // new "memory end" including padding
+                mstore(0x40, add(o_code, and(add(add(size, 0x20), 0x1f), not(0x1f))))
+                // store length in memory
+                mstore(o_code, size)
+                // actually retrieve the code, this needs assembly
+                extcodecopy(100700, add(o_code, 0x20), 0, size)
```

Comment:
you don't care about the output buffer, so 
`extcodecopy(addr, 0, 0, 2)`
would be enough.
(you have "scratch storage" of 64 bytes from offset zero. That's why the "free mem memory ptr" is at mload(64)
also, your code does "extcodecopy" of size "zero". theoretically, an "assembly optimizer" can remove this opcode, which copies nothing.